src/uu/install/src/parse.rs

```diff
@@ -0,0 +1,312 @@
+        Ok(u) => Ok(u),
+        Err(_) => {
+            if let Ok(i) = user.parse() {
+                return Ok(i);
+            }
+
+            Err(InstallError::InvalidUser(user.to_string()).into())
+        }
+    }
+}
+
+#[derive(Arguments)]
+#[arguments(file = "install.md")]
+enum Arg {
+    /// make a backup of each existing destination file
+    #[arg("-b", "--backup[=CONTROL]", value=BackupMode::Existing)]
+    Backup(BackupMode),
+
+    /// (ignored)
+    #[arg("-c")]
+    C,
+
+    /// compare each pair of source and destination files, and in some cases, do not modify the destination at all
+    #[arg("-C", "--compare")]
+    Compare,
+
+    /// treat all arguments as directory names. create all components of the specified directories
+    #[arg("-d", "--directory")]
+    Directory,
+
+    /// create all leading components of DEST except the last, then copy SOURCE to DEST
+    #[arg("-D")]
+    CreateLeading,
+
+    /// unsupported (TODO) (implies verbose)
+    #[arg("--debug")]
+    Debug,
+
+    /// set group ownership, instead of process's current group
+    #[arg("-g GROUP", "--group=GROUP")]
+    Group(String),
+
+    /// set group ownership, instead of process's current group
+    #[arg("-m MODE", "--mode=MODE", value = Mode(0o755))]
+    Mode(Mode),
+
+    /// set ownership (super-user only)
+    #[arg("-o OWNER", "--owner=OWNER")]
+    Owner(String),
+
+    /// apply access/modification times of SOURCE files to corresponding destination files
+    #[arg("-p", "--preserve-timestamps")]
+    PreserveTimestamps,
+
+    /// strip symbol tables (no action Windows)
+    #[arg("-s", "--strip")]
+    Strip,
+
+    /// program used to strip binaries (no action Windows)
+    #[arg("--strip-program=PROGRAM")]
+    StripProgram(PathBuf),
+
+    /// program used to strip binaries (no action Windows)
+    #[arg("-S SUFFIX", "--suffix=SUFFIX")]
+    Suffix(String),
+
+    /// move all SOURCE arguments into DIRECTORY
+    #[arg("-t DIRECTORY", "--target-directory=DIRECTORY")]
+    TargetDirectory(PathBuf),
+
+    /// (unimplemented) treat DEST as a normal file
+    #[arg("-T", "--no-target-directory")]
+    NoTargetDirectory,
+
+    /// explain what is being done
+    #[arg("-v", "--verbose")]
+    Verbose,
+
+    /// "(unimplemented) preserve security context"
+    #[arg("--preserve-context")]
+    PreserveContext,
+
+    /// (unimplemented) set security context of files and directories
+    #[arg("-Z", "--context[=CTX]", value=None)]
+    Context(Option<OsString>),
+}
+
+impl Options<Arg> for Settings {
+    fn apply(&mut self, arg: Arg) {
+        match arg {
+            Arg::Backup(b) => self.backup_mode = b,
+            Arg::C => {}
+            Arg::Compare => self.compare = true,
+            Arg::Directory => self.main_function = MainFunction::Directory,
+            Arg::CreateLeading => self.create_leading = true,
+            Arg::Debug => {
+                self.verbose = true;
+                unimplemented!()
```

Comment:
Lol (it's correct, but still funny to set a value and then immediately panic)

Reply:
well, the docs say it's equivalent to setting verbose and then some. I thought I'd do the verbose, and not bother with the then some. Actually implementing this should be an issue

Reply:
Yeah I agree! I just thought it was funny :)